### PR TITLE
BE-1038

### PR DESCRIPTION
**Pull request by leonklinke**
## Change description

> Description here

## Type of change
- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request name and commit comments all contain JIRA user story

### CODE REVIEW

File: source/includes/bank_accounts.md [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+`routing_number` | string | Routing number 
+`account_type` | string | Account Type ("checking" "savings" "other")
+`bank_name` | string | Bank name
+`kind` | string | Kind of account ("internal" "virtual" "external")
+`created_at` | timestamp | Date of creation
+`updated_at` | timestamp | Date of update
+
+
+
+
```
Comment (recmend) [inline source/includes/bank_accounts.md:67]: Are these lines intentional?

File: source/includes/bank_accounts.md [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+`routing_number` | string | Routing number (masked)
+`account_type` | string | Account Type ("checking" "savings" "other")
+`bank_name` | string | Bank name
+`kind` | string | Kind of account ("internal" "virtual" "external")
+`created_at` | timestamp | Date of creation
+`updated_at` | timestamp | Date of update
+
+
+
+
```
Comment (recmend) [inline source/includes/bank_accounts.md:123]: Same comment

File: source/includes/bank_accounts.md [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+  -H "Content-Type: application/json"
+```
+
+> Response
+
+```json
+{
+  "id": "d938f36e-c4...",
+  "user_id": "c8e1f36e-c4...",
+  "user": {},
```
Comment (recmend) [inline source/includes/bank_accounts.md:139]: ideally, we return the user object here in the same response

File: source/includes/bank_accounts.md [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+  "created_at": 1629916884,
+  "updated_at": 1629916884
+}
+```
+
+This API returns a user's specified bank account based on the user id and bank account id.
+
+### HTTP Request
+
+`GET http://api.highline.co/users/{id}/bank_accounts/{id_account}`
```
Comment (recmend) [inline source/includes/bank_accounts.md:154]: `id_account` -> `bank_account_id`

File: source/includes/bank_accounts.md [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+
+### Parameters
+
+Name | Type | In | Description
+--------- | ------- | ------- | ------
+`id` *required* | string | path | User ID
+`id_account` *required* | string | path | Bank Account ID
```
Comment (recmend) [inline source/includes/bank_accounts.md:161]: same comment.

File: source/includes/bank_accounts.md [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+`routing_number` | string | Routing number (masked)
+`account_type` | string | Account Type ("checking" "savings" "other")
+`bank_name` | string | Bank name
+`kind` | string | Kind of account ("internal" "virtual" "external")
+`created_at` | timestamp | Date of creation
+`updated_at` | timestamp | Date of update
+
+
+
+
```
Comment (recmend) [inline source/includes/bank_accounts.md:180]: intentional?

File: source/includes/bank_accounts.md [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+{
+  "sucess": true
+}
+```
+
+This API updates a user bank account. By passing the bank account information you like, Highline will return a sucess message or an error.
+
+### HTTP Request
+
+`PUT http://api.highline.co/users/{id}/bank_accounts/{id_account}`
```
Comment (recmend) [inline source/includes/bank_accounts.md:208]: `bank_account_id`

File: source/includes/bank_accounts.md [hunk 8]
```diff
@@ -0,0 +1,328 @@
...
+`id_account` *required* | string | path | Bank account id
+`account_number` *optional* |string | body | Account number
+`routing_number` *optional* | string | body | Routing number
+`account_type` *optional* | string | body | Type of account (checking, savings, etc)
+`bank_name` *optional* | string | body | Name of bank
+
+
+
+
+
```
Comment (recmend) [inline source/includes/bank_accounts.md:224]: intentional?

File: source/includes/bank_accounts.md [hunk 9]
```diff
@@ -0,0 +1,328 @@
...
+{
+  "account_number": "54615646"
+}
+```
+
+This API display an account number of a user bank account. Highline will return the number, if you are the owner of that account.
+
+### HTTP Request
+
+`GET http://api.highline.co/users/{id}/bank_accounts/{id_account}/account_number`
```
Comment (recmend) [inline source/includes/bank_accounts.md:246]: `bank_account_id`

File: source/includes/bank_accounts.md [hunk 10]
```diff
@@ -0,0 +1,328 @@
...
+
+### Parameters
+
+Name | Type | In | Description
+--------- | ------- | ------- | ------
+`id` *required* | string | path | User id
+`id_account` *required* | string | path | Bank account id
```
Comment (recmend) [inline source/includes/bank_accounts.md:253]: same here

File: source/includes/bank_accounts.md [hunk 11]
```diff
@@ -0,0 +1,328 @@
...
+  "amount": 50000,
+  "time": 1629916884,
+}
+```
+
+This API display the balance of a user bank account. Highline will return the number.
+
+### HTTP Request
+
+`GET http://api.highline.co/users/{id}/bank_accounts/{id_account}/balance`
```
Comment (recmend) [inline source/includes/bank_accounts.md:279]: global PR feedback `id_account` -> `bank_account_id`

File: source/includes/bill_payments.md [hunk 12]
```diff
@@ -0,0 +1,267 @@
...
+        "updated_at": 1631304119
+      },
+      "colors": {
+        "primary": "",
+        "secondary": "",
+        "background": ""
+      },
+      "enrollments": null,
+      "link_keys": null,
+      "api_keys": null,
```
Comment (recmend) [inline source/includes/bill_payments.md:41]: we can hide these optional fields for now.

File: source/includes/bill_payments.md [hunk 13]
```diff
@@ -0,0 +1,267 @@
...
+      "bill_payment_id": "e87f74be-c46...",
+      "amount_expected": 50000,
+      "amount_paid": 0,
+      "paid": false,
+      "status": "",
+      "type": "",
+      "payment_order_id": "",
+      "created_at": 1631304126,
+      "updated_at": 1631304126
+  }],
```
Comment (recmend) [inline source/includes/bill_payments.md:79]: this actually doesn't belong here. all has_many resources should follow
parent/:id/children , i.e., /bill_payments/:id/transactions.
This is to make the route more performant and support paginations for has_many based relationships.

File: source/includes/bill_payments.md [hunk 14]
```diff
@@ -0,0 +1,267 @@
...
+  "employer_name": "Christian",
+  "logo_url": "https://cdn-icon....png",
+  "loan_description": "Consequatur accusantium ...",
+  "amount_paid": 100,
+  "amount_due": 100,
+  "ForecastedDueAmount": 0,
+  "ForecastedDueDate": 0,
+  "status": "paid",
+  "settlement_id": null,
+  "settlement": null,
```
Comment (recmend) [inline source/includes/bill_payments.md:89]: I would add example values.

File: source/includes/bill_payments.md [hunk 15]
```diff
@@ -0,0 +1,267 @@
...
+`end_date` *optional* | string | query | Paid at until date
+
+### Response
+
+Parameter | Type | Description
+--------- | ------- | -----------
+`id` | string | Bill Payment ID
+`enrollment_id` | string | Enrollment ID
+`enrollment` | json | Enrollment structure 
+`transactions` | array | Collection of transactions structure
```
Comment (recmend) [inline source/includes/bill_payments.md:119]: same comment, has_many relationships shouldn't be part of the response.

File: source/includes/bank_accounts.md (hunk 3 above)
Comment (leonklinke) [reply source/includes/bank_accounts.md:139]: in that particular case, we don't return users (no preload), so this object is going to be empty

Comment (recmend) [reply source/includes/bank_accounts.md:139]: Then, can we remove it here and add omitempty in the json.

**Pull request merged**